src/_pytest/config/__init__.py

```diff
@@ -152,6 +153,7 @@ def main(
     :returns: An exit code.
     """
     try:
+        os.environ["PYTEST_VERSION"] = __version__
```

Comment:
Let's make this more robust by preserving the previous value of `PYTEST_VERSION`, if any.

```suggestion
        old_pytest_version = os.environ.get("PYTEST_VERSION")
        os.environ["PYTEST_VERSION"] = __version__
```